druntime/src/importc.h

```diff
@@ -97,12 +97,17 @@ typedef unsigned long long __uint64_t;
  * __has_extension is a clang thing:
  *    https://clang.llvm.org/docs/LanguageExtensions.html
  * ImportC no has extensions.
+ *
+ * >=gcc-14 prints a warning for each #undef. This breaks a lot of
+ * things like the testsuite which expects clean output.
  */
+#if __GNUC__ < 14
```

Comment:
The comment says it's a clang thing, but I don't think clang defines `__GNUC__`. Does GCC even need this? The condition could just be 'if clang'.

Reply:
> The comment says it's a clang thing, but I don't think clang defines __GNUC__.

It seems clang does define `__GNUC__` which, by default, gets set to `4.2.1`: https://github.com/llvm/llvm-project/blob/6ca0f44cd89aa802c306c303764eabf83a7f5029/clang/lib/Driver/ToolChains/Clang.cpp#L7071

> Does GCC even need this? The condition could just be 'if clang'.

I'm fine with any condition that gets rid of the warning. My reason for testing against the specific gcc version is because I'd rather not change old behavior that people possibly depend on. However, looking more closely, gcc doesn't define the `__has_feature` or `__has_extension` macros and the commit which introduced them was specifically meant for clang.

Reply:
Looks like older gcc doesn't cope at all with `__has_feature`. gcc-14 doesn't seem to mind it even though it doesn't appear in its predefined macro list.

We could:
- revert to the original fix
- add an `#ifdef __clang__` to the failing test.